core/src/main/java/org/htsjdk/core/api/PathURI.java

```diff
@@ -0,0 +1,86 @@
+package org.htsjdk.core.api;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.nio.file.Path;
+
+/**
+ * Interface representing htsjdk-next input/output resources as URIs.
+ */
+public interface PathURI {
```

Comment:
As this will represent a resource, what's about `IOResource`? I'm not sure if it is better, but I don't like the `PathURI` name...

Reply:
Agreed, I don't love the names. IOResource might be an improvement - we'd still need a name for the default implementation class (currently called `PathDpecifier` for lack of a better name.

Reply:
Changed to IOResource; still not sure about PathSpecifer.

Reply:
Changed to IOResource; still not sure about PathSpecifier.